tools/ktx/command_extract.cpp

```diff
@@ -23,38 +23,150 @@
 
 /** @page ktxtools_extract ktx extract
 @~English
 
-Extracts a KTX2 file.
-
-@warning TODO Tools P5: This page is incomplete
+Extract selected images from a KTX2 file.
 
 @section ktxtools_extract_synopsis SYNOPSIS
-    ktx extract [options] @e input_file
+    ktx extract [option...] @e input_file @e output_path
 
 @section ktxtools_extract_description DESCRIPTION
+    @b ktx @b extract can extract one or multiple images from the KTX2 file specified as the
+    @e input_file argument and based on the format save them as RAW, EXR or PNG image files
+    to the @e output_path.
+    If the input file is invalid the first encountered validation error is displayed
+    to the stderr and the command exits with the relevant non-zero status code.
+
+    The @e output_path is interpreted as output filepath for single and output directory for
+    multi-image extracts.
+    When extracting multiple images with either '--all' or any of the 'all' args the
+    following naming is used for each output file:
+    <pre>output_path/output_level{}_face{}_layer{}_depth{}.extension</pre>
+    - Where the @e _level{} part is only present if the source texture has more than 1 level
+    - Where the @e _face{} part is only present if the source texture is cubemap or cubemap array (Cubemap)
+    - Where the @e _layer{} part is only present if the source texture is an array texture (Array)
+    - Where the @e _depth{} part is only present if the source texture baseDepth is more than 1 (3D)
+    - Where the @e {} is replaced with the numeric index of the given component starting from 0
+    - Where the @e extension part is "raw", "png" or "exr" based on the export format<br />
+    Note: The inclusion of the optional parts are determined by the source texture regardless of
+    which images are requested.
+
+    For non-raw exports the output image format is chosen to be the smallest related lossless
+    format:
+    - _UNORM formats exported as PNG 8 or 16 bit (except for Depth/Stencil)
+    - _SINT/_UINT formats exported as EXR with Half/Float/UInt
+    - _SFLOAT/_UFLOAT formats exported as EXR with Half/Float
 
     The following options are available:
     <dl>
-        <dt>-f, --flag</dt>
-        <dd>Flag description</dd>
+        <dt>--transcode &lt;target&gt;</dt>
+        <dd>Transcode the texture to the target format before executing the extract.
+            Requires the input file to be transcodable (it must be either BasisLZ
+            supercompressed or has UASTC color model in the DFD). This option matches the
+            functionality of the @ref ktxtools_transcode "ktx transcode" command.
+            If the target option is not set the r8, rg8, rgb8 or rgba8 target will be selected
+            based on the number of channels in the input texture.
+            Case-insensitive. Possible options are:
+            etc-rgb | etc-rgba | eac-r11 | eac-rg11 | bc1 | bc3 | bc4 | bc5 | bc7 | astc |
+            r8 | rg8 | rgb8 | rgba8
```

Comment:
Add a note saying block compressed transcode targets can only be saved in raw format.